debezium-connector-db2/src/main/java/io/debezium/connector/db2/Db2Connection.java

```diff
@@ -160,7 +160,7 @@ public void getChangesForTables(Db2ChangeTable[] changeTables, Lsn intervalFromL
 
         int idx = 0;
         for (Db2ChangeTable changeTable : changeTables) {
-            final String query = GET_ALL_CHANGES_FOR_TABLE.replace(STATEMENTS_PLACEHOLDER, changeTable.getCaptureInstance());
+            final String query = GET_ALL_CHANGES_FOR_TABLE.replace(STATEMENTS_PLACEHOLDER, Db2ObjectNameQuoter.quoteNameIfNecessary(changeTable.getCaptureInstance()));
```

Comment:
Instead of doing this quoting again and again for each usage, how about considering this when obtaining the schema of the table/capture instance and bake this in there?

Reply:
I agree that it is not the prettiest, but the problem (if I got you correctly) is that the quoting is only needed when using the table/schema name in the SQL statements, but not every time the name of the table/schema is used in the code. I didn't want to get into extending the relevant objects with getName()/getNameQuoted() functions.

Reply:
> but not every time the name of the table/schema is used in the code.

Ok.

>  I didn't want to get into extending the relevant objects with getName()/getNameQuoted() functions.

Why not though? In fact, overriding `getCaptureInstance()` in `Db2ChangeTable` would be the perfect place for doing so, storing the quoted name in a field.

Reply:
> overriding getCaptureInstance() in Db2ChangeTable would be the perfect place for doing so, storing the quoted name in a field.

ok, let me take a look at it.